File: Anzeige2/src/main/java/com/capgemini/capybench/Anzeige2/entity/Post.java
package com.capgemini.capybench.Anzeige2.entity;

import jakarta.annotation.Nullable;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Objects;
import java.util.Set;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Post {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private String title;

    @Column
    private String content;

    @Column
    private LocalDateTime createdDate;

    @Column
    private LocalDateTime expirationDate;

    @ManyToOne
    private Topic topic;

    @ManyToOne
    @Nullable
    private Subtopic subtopic;

    @ManyToOne
    private Person author;

    //subscribers
    @ManyToMany(
            mappedBy = "subscribedPosts",
            fetch = FetchType.LAZY)
    private Set<Person> people;

    @PrePersist
    private void prePersist() {
        setCreatedDate(LocalDateTime.now());
    }

    @Override
    public boolean equals(final Object o) {
        if (this == o) {
            return true;
        }
        if (o == null || getClass() != o.getClass()) {
            return false;
        }
        Post post = (Post) o;
        return Objects.equals(id, post.id) && Objects.equals(title,
                                                             post.title) && Objects.equals(
                content,
                post.content) && Objects.equals(createdDate, post.createdDate) && Objects.equals(
                expirationDate,
                post.expirationDate) && Objects.equals(topic, post.topic) && Objects.equals(subtopic,
                                                                                            post.subtopic) && Objects.equals(
                author,
                post.author);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, title, content, createdDate, expirationDate, topic, subtopic, author);
    }

}

File: Anzeige 2.0/src/domain/Post.ts
import { Person } from "./Person";
import { Subtopic } from "./Subtopic";
import { Topic } from "./Topic";

export class Post {
    id: number;
    title: string;
    content: string;
    createDate: Date;
    author: Person;
    expirationDate: Date;
    topic: Topic;
    subtopic: Subtopic | undefined;
    subscriberIds: number[];
    type: string = "post";

    constructor(id: number, title: string, 
                content: string, createDate: Date, 
                author: Person, expirationDate: Date, 
                subscriberIds: number[], topic: Topic, 
                subtopic?: Subtopic) {
        this.id = id;
        this.title = title;
        this.content = content;
        this.author = author;
        this.createDate = new Date(createDate);
        this.expirationDate = new Date(expirationDate);
        this.topic = topic;
        this.subtopic = subtopic;
        this.subscriberIds = subscriberIds;
        this.type = "post";
    }
}